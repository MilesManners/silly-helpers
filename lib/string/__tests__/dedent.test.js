const dedent = require('../dedent')

describe('dedent', () => {
  it('empty string', () => {
    const output = dedent('')
    expect(output).toMatchSnapshot()
  })

  it('no indent', () => {
    const output = dedent(`
test string without indentation
that wraps
onto multiple
lines to test`)
    expect(output).toMatchSnapshot()
  })

  it('has indent', () => {
    const output = dedent(`
            A string that gets so long you need to break it over
            multiple lines. Luckily dedent is here to keep it
            readable without lots of spaces ending up in the string
            itself.`).trim()
    expect(output).toMatchSnapshot()
  })

  it('varying indent', () => {
    const output = dedent(`
      Leading and trailing lines will be trimmed, so you can write something like
      this and have it work as you expect:
    
        * how convenient it is
        * that I can use an indented list
            - and still have it do the right thing
    
      That's all.
    `).trim()
    expect(output).toMatchSnapshot()
  })
})
