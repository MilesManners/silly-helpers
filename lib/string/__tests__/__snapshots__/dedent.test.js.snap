// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dedent empty string 1`] = `""`;

exports[`dedent has indent 1`] = `
"A string that gets so long you need to break it over
multiple lines. Luckily dedent is here to keep it
readable without lots of spaces ending up in the string
itself."
`;

exports[`dedent no indent 1`] = `
"
test string without indentation
that wraps
onto multiple
lines to test"
`;

exports[`dedent varying indent 1`] = `
"Leading and trailing lines will be trimmed, so you can write something like
this and have it work as you expect:
    
  * how convenient it is
  * that I can use an indented list
      - and still have it do the right thing
    
That's all."
`;
